### PR TITLE
Bug 1402062 - Switch thNotify api to have named options

### PR DESCRIPTION
**Pull request by imbstack**
This is to make it easier to catch changes in review and to be easier to understand in general.

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

File: ui/js/services/main.js [hunk 1]
```diff
@@ -128,17 +128,35 @@ treeherder.factory('thNotify', [
...
+                }
+
                 var maxNsNotifications = 5;
                 var notification = {
                     message: message,
                     severity: severity,
                     sticky: sticky,
-                    linkText: linkText,
-                    url: url,
+                    linkText: opts.linkText,
```
Comment (camd) [inline ui/js/services/main.js]: I know you were trying to change less code (which I appreciate.  :) ).  But I wonder if it'd be nicer to just clean up this bit with a tiny refactor in ES6:

    thNotify.notifications.unshift({ ...opts, message, severity, created: Date.now() };

File: ui/js/services/main.js [hunk 2]
```diff
@@ -128,17 +128,35 @@ treeherder.factory('thNotify', [
...
              */
-            send: function (message, severity, sticky, linkText, url) {
+            send: function (message, severity, opts) {
                 $log.debug("received message", message);
+                opts = opts || {};
                 severity = severity || 'info';
-                sticky = sticky || false;
+                var sticky = opts.sticky || false;
+
+                if (!_.isPlainObject(opts)) {
```
Comment (camd) [inline ui/js/services/main.js]: This is a very kind way of correcting the developer.  :)  But, tbh, I'd rather we just throw an exception here if ``opts`` is passed in wrong.  Then the dev can say, "wth?" and look up the syntax and will know what to do.  :)

Obviously would need to do this check at the beginning of the function.

**Comment by camd** [comment]
Oh, I forgot to mention that Ed just merged a change to our linting strategy that could affect this PR.  Would you also please rebase on master?  Thanks!!  :)

**Comment by imbstack** [comment]
I like all of those changes. Thanks!

**Pull request merged**